packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1184,7 +1212,10 @@ export class TransactionController extends BaseController<
     return Number(txReceipt.status) === 0;
   }
 
-  private async requestApproval(txMeta: TransactionMeta): Promise<AddResult> {
+  private async requestApproval(
+    txMeta: TransactionMeta,
+    { shouldShowRequest } = { shouldShowRequest: true },
```

Comment:
Should this match the defaulting in `processApproval` for consistency so:

`{ shouldShowRequest = true } = {}`

Reply:
Indeed that way is better. Pushed the changes.